include/cachemere/cache.hpp

```diff
@@ -341,6 +342,41 @@ bool Cache<K, V, I, E, SV, SK, TS>::compare_evict(const K& candidate_key, size_t
     }
 }
 
+template<class K, class V, template<class, class> class I, template<class, class> class E, class SV, class SK, bool TS>
+void Cache<K, V, I, E, SV, SK, TS>::insert_or_update(K&& key, V&& value, size_t key_size, size_t value_size)
+{
+    auto key_and_item = m_data.find(key);
+    if (key_and_item != m_data.end()) {
+        // Update.
+
+        if (value_size > key_and_item->second.m_value_size) {
+            // This item got bigger with the update.
+            m_current_size += value_size - key_and_item->second.m_value_size;
+        } else {
+            // This item got smaller with the update.
+            m_current_size -= key_and_item->second.m_value_size - value_size;
+        }
+
+        key_and_item->second.m_value      = std::move(value);
+        key_and_item->second.m_value_size = value_size;
+        key_and_item->second.m_total_size = key_size + value_size;
```

Comment:
should it be key_size + key_size + value_size; based on line 376 ? :thinking:

Reply:
Nope, this part is [correct](https://github.com/coveooss/cachemere/blob/feature/import-construction/include/cachemere/item.h#L27)!